### ISSUE TITLE
Add support for forcing a complete re-scan

### ISSUE THREAD

**Issue by jvoisin** [deluan/navidrome#130]
It would be nice to be able to force navidrome to rescan the music folder, while ignoring the timestamps.

Bonus point if this is made accessible to admin users via a button in the web interface.

**Comment by deluan** [comment]
I'll be work on it very soon™️

In the meantime, here are a couple of workarounds: 

1) Using [SQLite cmd line](https://sqlite.org/cli.html), run the following SQL command:
```
delete from property where id like 'LastScan%';
update media_file set updated_at = '0001-01-01';
```
This will force a full rescan for all files, when the next scan happens.

2) Whenever you want to force Navidrome to update a folder, `touch` the files in it with a command like this:
```
$ find R.E.M_ -name "*.mp3" -exec touch "{}" \;
```

Note: Keep `ND_SCANINTERVAL=1m` (1 minute) in your configuration, so new changes are quickly picked-up. Frequent scans are not too demanding on the server as the scan takes less than 2secs even with really large libraries.

----

I've being thinking about creating a command line tool to rescan the whole library or parts of it, while I don't create a UI for it.

Another (or additional) option would be to implement the [startScan](http://www.subsonic.org/pages/api.jsp#startScan) API method. That would even enable us to use the [beets' subsonic update plugin](https://beets.readthedocs.io/en/v1.4.8/plugins/subsonicupdate.html), that triggers rescans after changing the library (for those who use [beets](https://beets.io/), of course). Thoughts?

**Comment by madcatsu** [comment]
@deluan that beets integration idea is awesome. I don't add things to my library without it going through `beets import` first to ensure cover art is embedded, a cover.jpg is included in the album folder and fixing up metadata. So I would only need to trigger a library rescan on import or delete with beets.

So a big +1 from me 👍 

I would suggest  making `ND_SCANINTERVAL` an optional setting then if this is implemented. Some people may be adverse to scanning large file system trees on slow spinning disks regularly if they have very large libraries.

**Comment by deluan** [comment]
@madcatsu That's also my use case, `beets` for all my music. But keep in mind that Navidrome only uses the embedded album art, it does not read `cover.jpg` or any other image files.

By the way `ND_SCANINTERVAL` can be set to 0 to disable scan altogether.

**Comment by madcatsu** [comment]
> Navidrome only uses the embedded album art, it does not read `cover.jpg` or any other image files

That's fine, I've got those there for Plex support :)

**Comment by deluan** [comment]
There's now a `scan` command that can be issued from the command line:
```
$ navidrome scan [-f]
```
Use -f for scan all folders (not only the ones with detected changes)

I'll leaving this open, as it is still missing from the UI and API

**Comment by steifhahn** [comment]
How to use it in docker?
Version is V0.24

**Comment by deluan** [comment]
It will be available in V0.25.0. I'm still figuring out the details and will update the docs explaining how to use it.

Regarding the issues you are having, can you please open a ticket for that? It should not have missing songs.... even when there were some issues with the scan algorithm (that were fixed in #347)

**Comment by tcurdt** [comment]
Forcing a re-scan when running with `docker-compose` is:

```
docker-compose exec navidrome /app/navidrome scan -f
```

**Comment by deluan** [comment]
Subsonic API endpoints `startScan` and `getScanStatus` are now implemented in 0.37.0. `beets` integration is working (use the [subsonicupdate plugin](https://beets.readthedocs.io/en/stable/plugins/subsonicupdate.html)), and clients that support these endpoints can now trigger scans:


![ezgif-6-4a45132b37b0](https://user-images.githubusercontent.com/331353/98005868-89a23480-1dbf-11eb-8415-d60263268caa.gif)

**Comment by deluan** [comment]
I can finally close this, as the functionality is now fully implemented as of version 0.38.0 (both in the API and UI). To trigger a new rescan, use the new Activity Panel (top right, near the settings menu).
